core/matrix/bccoo_helper.hpp

```diff
@@ -0,0 +1,626 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright (c) 2017-2023, the Ginkgo authors
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#ifndef GKO_CORE_MATRIX_BCCOO_HELPER_HPP_
+#define GKO_CORE_MATRIX_BCCOO_HELPER_HPP_
+
+
+#include <cstring>
+#include <functional>
+
+
+#include <ginkgo/core/base/array.hpp>
+#include <ginkgo/core/base/types.hpp>
+
+
+#include "core/base/unaligned_access.hpp"
+#include "core/matrix/bccoo_aux_structs.hpp"
+
+
+namespace gko {
+namespace matrix {
+namespace bccoo {
+
+
+/*
+ *  Constants to manage bccoo objects
+ */
+
+
+#define WITH_LUT 1
+
+
+constexpr uint8 type_mask_value_is_in_lut = 0x80;
+constexpr uint8 type_mask_get_column_with_lut = 0x7F;
+constexpr uint8 type_mask_get_column_without_lut = 0xFF;
+
+#ifdef WITH_LUT
+
+constexpr uint8 cst_mark_end_row = 0xFF;
+constexpr uint8 cst_mark_size_big_row = 0x7E;
+constexpr uint8 cst_mark_size_medium_row = 0x7D;
+
+constexpr uint8 cst_max_size_small_idxs_row = 0xFF;
+constexpr uint8 cst_max_size_small_row = 0x7C;
+constexpr int cst_max_size_medium_row = 0xFFFF;
+
+#else
+
+constexpr uint8 cst_mark_end_row = 0xFF;
+constexpr uint8 cst_mark_size_big_row = 0xFE;
+constexpr uint8 cst_mark_size_medium_row = 0xFD;
+
+constexpr uint8 cst_max_size_small_idxs_row = 0xFF;
+constexpr uint8 cst_max_size_small_row = 0xFC;
+constexpr int cst_max_size_medium_row = 0xFFFF;
+
+#endif
```

Comment:
For test purposes, this is perfectly fine.
However, you should not merge this into `develop`. If you want to keep the macro (which I hope you don't as it is hard to test both paths), you need to rename it to avoid collisions (as namespaces don't apply to macros).

Reply:
I have temporary removed the macro. I will think if both definitions should be maintained or not.
Done!!